### ISSUE TITLE
FileID.get_vfd_handle() for non-SEC2 (core)

### ISSUE THREAD

**Issue by dnelson86** [h5py/h5py#550]
Hi,

Why does ```get_vfd_handle()``` raise NotImplemented for drivers other than SEC2? Is this something important I am missing?

I want to read the raw byte data corresponding to a HDF5 "file" which is generated in-memory only (core driver). Reference: http://lists.hdfgroup.org/pipermail/hdf-forum_lists.hdfgroup.org/2011-May/004691.html

My approach is roughly:
```
f = h5py.File('out.hdf5','w',driver='core',backing_store=False)
dset = f.require_dataset( 'test', (1000,), dtype='float32' )
dset[:] += 1.0
f.flush()

handle = f.fid.get_vfd_handle()
N = f.fid.get_filesize()

import ctypes
raw_bytes = (ctypes.c_char*N).from_address(handle)
with open('test.hdf5','wb') as f_out:
    f_out.write(raw_bytes)
```

I would love some opinions if I am doing something idiotic, and/or if I can do this without modifying the h5py source and recompiling.

**Comment by dnelson86** [comment]
Or (excuse my ignorance), can I make the call

```
H5Fget_vfd_handle(self.id, H5Fget_access_plist(self.id), <void**>&handle)
```

in my user code somehow?

**Comment by andrewcollette** [comment]
I'm not exactly sure... I've not interacted with the VFD code myself.  I don't believe there's a way to make that second snippet work directly from Python, though.

You could ask on the mailing list (h5py at google groups) to see if anyone's run into this before.

**Comment by dnelson86** [comment]
What about changing
```
if H5Pget_driver(H5Fget_access_plist(self.id)) != h5fd.SEC2:
```
to
```
if H5Pget_driver(H5Fget_access_plist(self.id)) not in [h5fd.SEC2,h5fd.CORE]:
```
With this change, the first code snippet does appear to work ok.

**Comment by andrewcollette** [comment]
I recall there were some newer functions added to HDF5 specifically to get/set the binary contents of CORE files.  You might double-check to see if one of those fits your needs.  A PR for those (or your above change) with tests would likely be accepted.

**Comment by dnelson86** [comment]
Hi,

Just one more note for anyone who might find this later: ran into a strange seg fault when using h5py within a Django application in mod_wsgi (apache daemon mode). Only when running in a daemon process spawned from httpd I found that "f.fid.get_vfd_handle()" returned negative values. I ended up modifying h5f.pyx again and changing "cdef int *handle" to "cdef int64_t *handle", which solved the issue (the memory addresses returned were e.g. 139864087163200 > 32bit).
